modules/core/src/main/scala/lucuma/odb/api/model/SpatialProfileModel.scala

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model
+
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import lucuma.core.util.Enumerated
+import lucuma.core.util.Display
+
+object SpatialProfileModel {
+  final case class Input(sourceType: SpatialProfileType, fwhm: Option[GaussianSourceAngleInput]) extends Product with Serializable
+
+  sealed abstract class Units(
+    val angleUnit: AngleModel.Units
+  ) extends Product with Serializable
+
+  object Units {
+
+    case object Microarcseconds extends Units(AngleModel.Units.Microarcseconds)
+    case object Milliarcseconds extends Units(AngleModel.Units.Milliarcseconds)
+    case object Arcseconds      extends Units(AngleModel.Units.Arcseconds)
+
+    val microarcseconds: Units = Microarcseconds
+    val milliarcseconds: Units = Milliarcseconds
+    val arcseconds: Units      = Arcseconds
+
+    implicit val EnumeratedUnits: Enumerated[Units] =
+      Enumerated.of(
+        Microarcseconds,
+        Milliarcseconds,
+        Arcseconds
+      )
+
+    implicit def DisplayUnits: Display[Units] =
+      Display.byShortName(_.angleUnit.abbreviation)
+
+  }
+
+  final case class GaussianSourceAngleInput(
+    microarcseconds: Option[Long],
+    milliarcseconds: Option[BigDecimal],
+    arcseconds:      Option[BigDecimal],
+    fromLong:        Option[NumericUnits.LongInput[Units]],
+    fromDecimal:     Option[NumericUnits.DecimalInput[Units]]
+  )
```

Comment:
Here is where I'd typically place something like

```scala
{
  def toAngle: ValidatedInput[Angle] =
    ValidatedInput.requireOne("fwhm",
      microarcseconds.map(Microarcseconds.readLong),
      milliarcseconds.map(Milliarcseconds.readDecimal),
      arcseconds.map(Arcseconds.readDecimal),
      fromLong.map(_.read),
      fromDecimal.map(_.read)
    )
}
```

Reply:
I guess that doesn't affect the graphql side right? You'd only use e.g. when executing the query?

Reply:
Right, it would be used as part of validating the input and turning it into the model type you use internally for the matching.